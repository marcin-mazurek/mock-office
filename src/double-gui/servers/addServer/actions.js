export const INIT = 'addServer/INIT';

export default (name, port, serverType, isSecure, keyPath, certPath) => ({
  type: INIT,
  name,
  port,
  serverType,
  isSecure,
  keyPath,
  certPath
});
