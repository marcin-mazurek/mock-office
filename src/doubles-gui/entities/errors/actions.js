export const ADD = 'errors/ADD';
export const REMOVE = 'errors/REMOVE';

export const add = reason => ({
  type: ADD,
  reason
});

export const remove = id => ({
  type: REMOVE,
  id
});
