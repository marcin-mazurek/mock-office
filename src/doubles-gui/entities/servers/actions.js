export const ADD = 'servers/ADD';
export const SELECT = 'servers/SELECT';
export const START = 'servers/START';
export const STOP = 'servers/STOP';
export const REMOVE = 'servers/REMOVE';
export const RENAME = 'servers/RENAME';

export const add = (name, port, serverType, id, secure) => ({
  type: ADD,
  name,
  port,
  id,
  serverType,
  secure
});

export const select = id => ({
  type: SELECT,
  id
});

export const start = id => ({
  type: START,
  id
});

export const stop = id => ({
  type: STOP,
  id
});

export const remove = id => ({
  type: REMOVE,
  id
});

export const rename = (id, name) => ({
  type: RENAME,
  id,
  name
});
