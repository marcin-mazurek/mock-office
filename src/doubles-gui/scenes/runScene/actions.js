export const RUN = 'scenes/RUN';
export const STOP = 'scenes/STOP';

export const run = (serverId, taskId) => ({
  type: RUN,
  serverId,
  taskId
});

export const stop = (serverId, taskId) => ({
  type: STOP,
  serverId,
  taskId
});
