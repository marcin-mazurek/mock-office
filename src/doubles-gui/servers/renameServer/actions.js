export const INIT = 'renameServer/INIT';

export const init = (id, name) => ({
  type: INIT,
  id,
  name
});
