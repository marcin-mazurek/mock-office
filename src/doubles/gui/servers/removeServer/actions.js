export const INIT = 'removeServer/INIT';

export const init = id => ({
  type: INIT,
  id
});
