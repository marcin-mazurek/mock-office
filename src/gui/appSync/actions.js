export const RESTORE_STATE = 'appSync/RESTORE_STATE';
export const restoreStateAction = servers => ({
  type: RESTORE_STATE,
  servers
});
