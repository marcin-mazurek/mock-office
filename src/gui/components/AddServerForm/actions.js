export const SUBMIT_BUTTON_CLICKED = 'addServerForm/SUBMIT_BUTTON_CLICKED';
export const submitButtonClickAction = values => ({
  type: SUBMIT_BUTTON_CLICKED,
  values
});
