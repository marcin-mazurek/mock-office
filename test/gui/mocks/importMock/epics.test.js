import { Map } from 'immutable';
import configureMockStore from 'redux-mock-store';
import 'rxjs';
import { createEpicMiddleware } from 'redux-observable';
import { initAction } from '../../../../src/gui/mocks/importMock/actions';

describe('importMockEpic', () => {
  let store;

  afterEach(() => {
    store.clearActions();
  });

  test.only('snapshot of success flow', (done) => {
    jest.mock('../../../../src/gui/api', () => ({
      addMock() {
        return Promise.resolve({
          data: {
            id: 'mock-id',
            tasks: []
          }
        });
      }
    }));
    const addMockFromFileEpic = require('../../../../src/gui/mocks/importMock/epics').default;
    const epicMiddleware = createEpicMiddleware(addMockFromFileEpic);
    const mockStore = configureMockStore([epicMiddleware]);
    store = mockStore(new Map({
      mocks: new Map({
        entities: new Map({
          'mock-id': {
            type: 'server type',
            name: 'server name',
            port: 3000,
            id: 'mock-id'
          }
        })
      })
    }));

    store.dispatch(initAction('server-id', 'scenario-id', [
      new Blob(['[{"tasks":[]}]'])
    ]));
    setTimeout(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    }, 100);
  });
});
