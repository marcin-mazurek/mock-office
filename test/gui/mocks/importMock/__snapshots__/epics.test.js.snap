// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importMockEpic snapshot of success flow 1`] = `
Array [
  Object {
    "files": Array [
      Blob {
        Symbol(impl): BlobImpl {
          "_buffer": Object {
            "data": Array [
              91,
              123,
              34,
              116,
              97,
              115,
              107,
              115,
              34,
              58,
              91,
              93,
              125,
              93,
            ],
            "type": "Buffer",
          },
          "isClosed": false,
          "type": "",
          Symbol(wrapper): [Circular],
        },
      },
    ],
    "scenario": "scenario-id",
    "server": "server-id",
    "type": "importMock/INIT",
  },
  Object {
    "id": "mock-id",
    "params": Object {
      "id": "mock-id",
      "tasks": Array [],
    },
    "scenario": "scenario-id",
    "type": "entities/ADD_MOCK",
  },
]
`;
